pkg/controller/workload/workload_processor.go

```diff
@@ -654,20 +657,56 @@ func (p *Processor) handleAuthorizationTypeResponse(rsp *service_discovery_v3.De
 			return err
 		}
 		policyKey := authPolicy.ResourceName()
-		if err := auth.FlushAuthzMap(p.hashName.StrToNum(policyKey), authPolicy); err != nil {
-			return fmt.Errorf("flushAuthz %s failed %v ", policyKey, err)
+		if err := maps_v2.AuthorizationUpdate(p.hashName.Hash(policyKey), authPolicy); err != nil {
+			return fmt.Errorf("AuthorizationUpdate %s failed %v ", policyKey, err)
 		}
 	}
 
 	// delete resource by name
 	for _, resourceName := range rsp.GetRemovedResources() {
 		rbac.RemovePolicy(resourceName)
+		if err := maps_v2.AuthorizationDelete(p.hashName.Hash(resourceName)); err != nil {
+			log.Errorf("remove authorization policy %s failed :%v", resourceName, err)
+		}
 		log.Debugf("remove authorization policy %s", resourceName)
 	}
 
+	p.handleRemovedAuthzPolicyDuringRestart(rbac)
 	return nil
 }
 
+// When processing the Authorization's response for the first time,
+// fetch the data from the /mnt/workload_hash_name.yaml file
+// and compare it with the data in the cache.
+func (p *Processor) handleRemovedAuthzPolicyDuringRestart(rbac *auth.Rbac) {
+	var (
+		policyValue = security_v2.Authorization{}
+	)
+
+	if kmeshbpf.GetStartType() != kmeshbpf.Restart {
+		return
+	}
+
+	log.Infof("reload authz config from last epoch")
+	/* We traverse hashName, if there is a record exists in bpf map
+	 * but not in usercache, that means the data in the bpf map load
+	 * from the last epoch is inconsistent with the data that should
+	 * actually be stored now. then we should delete it from bpf map
+	 */
+	policyCache := rbac.GetAllPolicies()
+	for str, num := range p.hashName.strToNum {
+		if _, exists := policyCache[str]; !exists {
+			log.Debugf("policyCache[%v] not exists", str)
```

Comment:
This log may be printed many times. Is it directly printed to L701?

Reply:
fix